src/gluonts/transform/convert.py

```diff
@@ -759,3 +759,67 @@ def _empirical_cdf_inverse_transform(
             tensor_to_numpy(input_batch["intercepts"]),
         )
     return outputs
+
+
+class SparseToDense(FlatMapTransformation):
+    """
+    Convert a sparse univariate time series to the `interval-size` format,
+    i.e., a two dimensional time series where the first dimension corresponds
+    to the time since last positive value (1-indexed), and the second dimension
+    corresponds to the size of the demand. This format is used often in the
+    intermittent demand literature, where predictions are performed on this
+    "dense" time series, e.g., as in Croston's method.
+
+    As an example, the time series `[0, 0, 1, 0, 3, 2, 0, 4]` is converted into
+    the 2-dimensional time series `[[3, 2, 1, 2], [1, 3, 2, 4]]`, with a
+    shape (2, M) where M denotes the number of non-zero items in the time series.
+
+    Parameters
+    ----------
+    target_field
+        The target field to be converted, containing a univariate and sparse
+        time series
+    drop_empty
+        If True, all-zero time series will be dropped.
+    discard_first
+        If True, the first element in the converted dense series will be dropped,
+        replacing the target with a (2, M-1) tet instead. This can be used
+        when the first 'inter-demand' time is not well-defined. e.g., when the true
+        starting index of the time-series is not known.
+    """
+
+    @validated()
+    def __init__(
+        self,
+        target_field: str,
+        drop_empty: bool = False,
+        discard_first: bool = False,
+    ) -> None:
+
+        self.target_field = target_field
+        self.drop_empty = drop_empty
+        self.discard_first = discard_first
+
+    def _process_sparse_time_sample(self, a: List) -> Tuple[List, List]:
+        a = np.array(a)
+        nz = np.nonzero(a)[0]
+
+        if len(nz) == 0:
+            return [], []
+
+        times = np.diff(np.r_[-1, np.nonzero(a)[0]]).astype(np.float).tolist()
+        sizes = a[np.nonzero(a)].tolist()
+
+        return (times[1:], sizes[1:]) if self.discard_first else (times, sizes)
```

Comment:
I think this is easier written as proper statements

```suggestion
        if self.discard_first:
            return times[1:], sizes[1:]

        return times, sizes
```